pennylane/interfaces/jax_jit.py

```diff
@@ -73,10 +73,15 @@ def execute(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_d
     if max_diff > 1:
         raise InterfaceUnsupportedError("The JAX interface only supports first order derivatives.")
 
-    for tape in tapes:
-        # set the trainable parameters
-        params = tape.get_parameters(trainable_only=False)
-        tape.trainable_params = qml.math.get_trainable_indices(params)
+    if _n == 1:
+        for tape in tapes:
+            # set the jitted parameters
+            params = tape.get_parameters(trainable_only=False)
+            trainable_params = set()
+            for idx, p in enumerate(params):
+                if isinstance(p, jax.core.Tracer) or qml.math.requires_grad(p):
+                    trainable_params.add(idx)

```

Comment:
I believe `trainable_params` is supposed to be a `list`, because we require it to maintain order (and be sorted).

```suggestion
            trainable_params = []
            for idx, p in enumerate(params):
                if isinstance(p, jax.core.Tracer) or qml.math.requires_grad(p):
                    trainable_params.append(idx)

```

Reply:
I took this from the math module where we define the following function:

```
@multi_dispatch(argnum=[0], tensor_list=[0])
def get_trainable_indices(values, like=None):
    trainable_params = set()

    for idx, p in enumerate(values):
        if requires_grad(p, interface=like):
            trainable_params.add(idx)

    return trainable_params
```

Reply:
:thinking: